src/server/tiered_storage.h

```diff
@@ -51,15 +56,23 @@ class TieredStorage {
 
   std::error_code Read(size_t offset, size_t len, char* dest);
 
+  bool AllowWrites() const;
```

Comment:
maybe: `IoDeviceOverloaded()` with the opposite meaning? AllowWrites is a too general name